src/puppetlabs/puppetdb/cli/export.clj

```diff
@@ -229,15 +210,25 @@
          :puppetlabs.kitchensink.core/cli-error (System/exit 1)
          :puppetlabs.kitchensink.core/cli-help  (System/exit 0))))))
 
-(defn -main
+(defn- main
   [& args]
-  (let [[{:keys [outfile host port]} _] (validate-cli! args)
-        nodes (get-nodes host port)]
+  (let [[{:keys [outfile host port url-prefix] :as opts} _] (validate-cli! args)
+        src {:protocol "http" :host host :port port :prefix url-prefix}
+        _ (if (utils/report-bad-base-url src "Invalid source")
+            (throw+ {:type :exit-process :status 1}))
```

Comment:
I think exit-process should be something different from :type. We usually use type to indicate the type of error that occurred, rather than what do do about the error. So in this case maybe invalid-source is the error type? I also prefer the namespace keywords in these potentially global maps, so we know where it came from, i.e. ::invalid-source.